engine/apps/auth_token/auth.py

```diff
@@ -264,6 +270,11 @@ class SlackTokenAuthentication(_SocialAuthTokenAuthentication[SlackAuthToken]):
     model = SlackAuthToken
 
 
+class MattermostTokenAuthentication(_SocialAuthTokenAuthentication[MattermostAuthToken]):
```

Comment:
Not needed right now, but how hard would be to have all mattermost-related model/logic in a separate app?

Reply:
Its not a hard thing to do. My thought process was since we are using the common OAuth view to complete the OAuth flow. We can keep the auth model alone in the `auth_token` folder along with other utilities like crypto, exceptions etc. The other models and class will be inside the mattermost app.